src/components/RecoveryPortalWebView/RecoveryPortalWebView.js

```diff
@@ -0,0 +1,109 @@
+// @flow
+/*
+    Pillar Wallet: the personal data locker
+    Copyright (C) 2019 Stiftung Pillar Project
+    This program is free software; you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation; either version 2 of the License, or
+    (at your option) any later version.
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+    You should have received a copy of the GNU General Public License along
+    with this program; if not, write to the Free Software Foundation, Inc.,
+    51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+*/
+import * as React from 'react';
+import { WebView } from 'react-native-webview';
+import { RECOVERY_PORTAL_URL } from 'react-native-dotenv';
+import { useBackHandler } from '@react-native-community/hooks';
+import type { NavigationScreenProp } from 'react-navigation';
+
+// components
+import { Wrapper } from 'components/Layout';
+import ContainerWithHeader from 'components/Layout/ContainerWithHeader';
+import Spinner from 'components/Spinner';
+
+
+type Props = {
+  onWebViewMessage?: Function,
+  navigation: NavigationScreenProp,
+  onWebViewNavigate?: Function,
+  onRef?: Function,
+  title?: string,
+  isVisible?: boolean,
+  webViewRef?: WebView,
+  urlPath?: string,
+  goBackDismiss?: boolean,
+};
+
+const renderLoading = () => <Spinner style={{ alignSelf: 'center', position: 'absolute', top: '50%' }} />;
+
+const RecoveryPortalWebView = ({
+  onWebViewMessage,
+  title,
+  onWebViewNavigate,
+  isVisible = true,
+  navigation,
+  onRef,
+  urlPath = '',
+  goBackDismiss,
+}: Props) => {
+  let webViewRef;
+  let canWebViewNavigateBack = false;
+  const uri = `${RECOVERY_PORTAL_URL}/${urlPath}`;
+
+  const handleNavigationBack = () => {
+    if (!webViewRef || !canWebViewNavigateBack) {
+      if (goBackDismiss) {
+        navigation.dismiss();
+        return;
+      }
+      navigation.goBack();
+      return;
+    }
+    webViewRef.goBack();
+  };
+
+  useBackHandler(() => {
+    handleNavigationBack();
+    return true;
+  });
+
+  const onNavigationStateChange = (webViewNavigationState) => {
+    canWebViewNavigateBack = !!webViewNavigationState.canGoBack;
+    if (onWebViewNavigate) onWebViewNavigate(webViewNavigationState);
+  };
+
+  return (
+    <ContainerWithHeader
+      headerProps={{
+        centerItems: [{ title: title || 'Recovery Portal' }],
+        customOnBack: handleNavigationBack,
+      }}
+    >
+      <Wrapper style={{ flex: 1 }}>
```

Comment:
Minor thing:
What if you'd center wrapper's content?
Then there won't be any need to make Spinner's position absolute.
(Now it's top position will be slightly off the center (as you'd need to transform y or add negative top margin to put it the center).

Reply:
Makes sense, BUT it would work for the general spinner, but won't work for WebView component spinner as prop, but I'll try to adjust by passing additional value to spinner render method.

Reply:
Actually it becomes more complex and requires more code lines (because I have to additionally dismiss centring on wrapper because of WebView component) so I would like to keep it the way it is, hope it makes sense.